### ISSUE TITLE
read_gml does not retrieve a list

### ISSUE THREAD

**Issue by MartinPJorge** [networkx/networkx#3511]
Hi,

`read_gml` fails when reading a node property that is a single element list.
Instead of reading a list of a single element, it considers that property as a single element. See the example below:

```python
import networkx as nx
g = nx.Graph()
g.add_node('n1', properties=['a'])
nx.write_gml(g, '/tmp/test.gml')
read_g = nx.read_gml('/tmp/test.gml')
type(read_g.nodes['n1']['properties'])
>>> <class 'str'>
```
Property's type should have been `list`, not type `str`.

**Comment by dschult** [comment]
This looks like a bug related to how the node attributes are formed.
The GML code doesn't actually store lists. Writing a node property that happens to be a list 
produces GML code with repeated lines for the same property.
When reading in the file, if there is one value for a property, how can we tell whether it should be created as a list or simply take on the single value?

    In [390]: import networkx as nx
     ...: g = nx.Graph()
     ...: g.add_node('n1', properties=['a','b'])
     ...: nx.write_gml(g, '/tmp/test.gml')
     ...: read_g = nx.read_gml('/tmp/test.gml')
     ...: type(read_g.nodes['n1']['properties'])
     ...: 
     ...: 
    Out[390]: list

    In [391]: !cat /tmp/test.gml
    graph [
      node [
        id 0
        label "n1"
          properties "a"
          properties "b"
      ]
    ]

    In [392]: import networkx as nx
     ...: g = nx.Graph()
     ...: g.add_node('n1', properties=['a'])
     ...: nx.write_gml(g, '/tmp/test.gml')
     ...: read_g = nx.read_gml('/tmp/test.gml')
     ...: type(read_g.nodes['n1']['properties'])
     ...: 
     ...: 
    Out[392]: str

    In [393]: !cat /tmp/test.gml
    graph [
      node [
        id 0
        label "n1"
          properties "a"
      ]
    ]

**Comment by MartinPJorge** [comment]
Hi,

I saw that misbehaviour in the code. I'd say that the best is to refactor the `write_gml` method so it's possible to store lists. According to [[1]](http://www.uni-passau.de/fileadmin/files/lehrstuhl/brandenburg/projekte/gml/gml-technical-report.pdf) GML lists should satisfy following syntax:
```BNF
List::=(whitespace* Key whitespace+ Value)*
Value::=Integer | Real | String | [ List ]
```
Thus, `write_gml` should produce something like this:
```python
import networkx as nx
g = nx.Graph()
g.add_node('n1', properties=['a', 'b'])
nx.write_gml(g, '/tmp/test.gml')
```
```GML
graph [
  node [
    id 0
    label "n1"
    properties [
      0 "a"
      1 "b"
    ]
  ]
]

```
Such refactoring is little effort and would make the function consistent with GML definition. What do you think?

### References
[1] Himsolt, M. (1997). GML: A portable graph file format. Html page under http://www. fmi. uni-passau. de/graphlet/gml/gml-tr. html, Universität Passau.

**Comment by dschult** [comment]
Note that this description of "Lists" is more closely aligned in Python with a description of dicts.
The key/value pairings indicate a dict. Also, the portion of the spec you reference does not include a spec for "Key". What are the allowable types for a Key?

**Comment by AnjoMan** [comment]
I found the spec that referenced by @MartinPJorge. based on what it says, i don't think the suggested format is even valid GML, since keys are supposed to start with alpha characters. I think
![image](https://user-images.githubusercontent.com/2414360/66880308-56142280-ef90-11e9-9c9d-3e260f831ebf.png). I tried loading a sample into gephi, another graph tool, and it couldn't even handle the file. the spec. The spec does consider writing lists, but it only uses repeated keys, which of course does not support the above-mentioned edge-case 
![image](https://user-images.githubusercontent.com/2414360/66880555-6aa4ea80-ef91-11e9-8331-7f094400e052.png)
